### PR TITLE
Add Chroma modular pipeline (text2image)

### LINKED ISSUE #13295: Modular Diffusers 🧨
Hey folks 👋 

We recently released [Modular Diffusers](https://huggingface.co/blog/modular-diffusers), which gives developers the flexibility to reuse existing pipeline blocks in different workflows and also easily implement custom "modular" blocks.

While the `DiffusionPipeline` class has helped establish a standard abstraction layer, it has also been a bit rigid when it comes to extensions. With Modular Diffusers, we want to give our users the utmost flexibility while also complementing our standard pipeline interfaces.

To improve modular, we are soliciting contributions from the community. Contributions for modular can be manifold:

* Implement a modular version of an existing pipeline
* Implement a cool custom modular pipeline and share it with us
* Identify a feature that could be improved in modular and improve it

We're looking forward to hearing from you!

### PR DESCRIPTION
**Pull request by charchit7**
# What does this PR do?

Fixes #13295. Adds a ModularPipeline for Chroma (t2i only):
- T5-only text encoder block with Chroma's padding-token attention masking
- guider-based CFG denoise loop (ClassifierFreeGuidance)
- reuses Flux-style packed-latent prep, mu-shift timesteps, RoPE ids, decode

Ran the self-review:
- Verdict: READY

## Before submitting
- [x] Did you use an AI agent (Claude Code, Codex, Cursor, etc.) to help with this PR? If so:
  - [x] Did you read the [Coding with AI agents](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#coding-with-ai-agents) guide?
  - [x] Did you run the [`self-review`](https://github.com/huggingface/diffusers/blob/main/.ai/skills/self-review/SKILL.md) skill on the diff?
  - [x] Did you share the final self-review notes in the PR description or a comment?
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution)?
- [x] Did you read our [philosophy doc](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)? (important for complex PRs)
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.

## Who can review?

@yiyixuxu @dg845 and @asomoza

### CODE REVIEW

**Comment by charchit7** [comment]
config:
~~~
import os
os.environ["HF_HOME"] = "/home/jupyter/images/hf_cache"

import numpy as np
import torch
from PIL import Image

from diffusers import ChromaPipeline
from diffusers.guiders import ClassifierFreeGuidance
from diffusers.modular_pipelines import ChromaAutoBlocks

MODEL_ID = "lodestones/Chroma1-HD"

PROMPT = (
    "A painting of a squirrel eating a burger, highly detailed, vibrant colors, "
    "professional art, sharp focus, cinematic lighting"
)

NEGATIVE_PROMPT = (
    "low quality, ugly, unfinished, out of focus, deformed, "
    "disfigure, blurry, smudged, restricted palette, flat colors"
)

HEIGHT = WIDTH = 1024
NUM_STEPS = 40
GUIDANCE_SCALE = 3.0
MAX_SEQ_LEN = 512

OUTPUT_DIR = "/home/jupyter/images/chroma_output"
os.makedirs(OUTPUT_DIR, exist_ok=True)

DTYPE = torch.bfloat16
DEVICE = "cuda"
SEED = 42
~~~

Modular output :
image

standard output:
image
